<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MazeGui</class>
 <widget class="QMainWindow" name="MazeGui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>729</width>
    <height>560</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string> MAZES AND CAVES</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>535</x>
      <y>18</y>
      <width>181</width>
      <height>507</height>
     </rect>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="mazePage">
     <attribute name="title">
      <string>MAZE</string>
     </attribute>
     <widget class="QPushButton" name="findBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>255</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>FIND WAY</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saveBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>141</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>SAVE MAZE</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="rowsSpin">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="value">
       <number>25</number>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>165</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="startBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>141</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Start point</string>
      </property>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>300</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QPushButton" name="loadBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>141</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>LOAD MAZE</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelRows">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>54</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>ROWS:</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelCols">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>70</y>
        <width>71</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>COLUMNS:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="genBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>GENERATE MAZE</string>
      </property>
     </widget>
     <widget class="QPushButton" name="endBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>220</y>
        <width>141</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Finish point</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="colsSpin">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>90</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="value">
       <number>25</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="cavePage">
     <attribute name="title">
      <string>CAVE</string>
     </attribute>
     <widget class="QPushButton" name="caveGenBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>220</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>INIT START FIELD</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="rowsSpin_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="value">
       <number>35</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelRows_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>54</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>ROWS:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="colsSpin_2">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>90</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="value">
       <number>35</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelCols_2">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>70</y>
        <width>71</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>COLUMNS:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="saveCaveBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>141</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>SAVE CAVE</string>
      </property>
     </widget>
     <widget class="QPushButton" name="loadCaveBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>141</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>LOAD CAVE</string>
      </property>
     </widget>
     <widget class="QPushButton" name="nextStepBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>290</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>NEXT STEP</string>
      </property>
     </widget>
     <widget class="QPushButton" name="autoIntervalBtn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>435</y>
        <width>141</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>RUN TIMELAPSE</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="stepSpin">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>400</y>
        <width>71</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>100</number>
      </property>
      <property name="maximum">
       <number>3000</number>
      </property>
      <property name="singleStep">
       <number>50</number>
      </property>
      <property name="value">
       <number>200</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelStep">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>380</y>
        <width>141</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>STEP, milliseconds:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="bornSpin">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>4</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelBorn">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>BORN LIMIT:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="deathSpin">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>140</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>4</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelDeath">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>120</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>DEATH LIMIT:</string>
      </property>
     </widget>
     <widget class="Line" name="line_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>250</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="labelStep_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>270</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>STEP BY STEP GENERATION</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelStep_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>350</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>TIMELAPSE GENERATION</string>
      </property>
     </widget>
     <widget class="Line" name="line_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>330</y>
        <width>141</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="chanceSpin">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>141</width>
        <height>26</height>
       </rect>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelChance">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>141</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>START INIT CHANCE, %:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="stopBtn">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>400</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="text">
       <string>STOP</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="statusLine">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>530</y>
      <width>691</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
